balancesExporter/export/exporter.go

```diff
@@ -1,43 +1,59 @@
 package export
 
 import (
+	"encoding/hex"
+	"encoding/json"
 	"fmt"
 	"io/ioutil"
 
 	"github.com/ElrondNetwork/elrond-go-core/core"
 	"github.com/ElrondNetwork/elrond-go-core/data"
+	"github.com/ElrondNetwork/elrond-go/sharding"
 	"github.com/ElrondNetwork/elrond-go/state"
 )
 
 // ArgsNewExporter holds arguments for creating an exporter
 type ArgsNewExporter struct {
-	TrieWrapper      trieWrapper
-	Format           string
-	Currency         string
-	CurrencyDecimals uint
-	WithContracts    bool
-	WithZero         bool
+	TrieWrapper             trieWrapper
+	Format                  string
+	WithProjectedShard      uint32
```

Comment:
could have used a component or at least a func handler in order to avoid passing 2 things: the value and the isSet flag

Reply:
or maybe the flags config struct as a whole?

Reply:
Fixed by defining an `OptionalUint32` struct (the flags struct contains some fields that are not needed by the exporter).